app/createTournament/createTournamentDialog.js

```diff
@@ -0,0 +1,21 @@
+import * as tournamentValidation from './validateTournament.js';
+import * as tournamentBusiness from '../tournamentBusiness.js'
+function showOrHideErrorMessage(validationResult) {
+	var $errorSummary = $('#errorsummary');
+	if (validationResult.error===null) {
+		$errorSummary.hide('slow');
+	} else {
+		var message = tournamentValidation.getErrorMessage(validationResult.error);
+		$errorSummary[0].innerText = message;
+		$errorSummary.show('slow');
+	}
+}
+export function createTournamentDialog(){
+ var listofPlayers=tournamentBusiness.getListofPlayers();
+ var validationResult=tournamentValidation.validateNumberofPlayers(listofPlayers.length);
+ showOrHideErrorMessage(validationResult);
+ if(validationResult.error===null ||validationResult.error.lessthanrequired===undefined){
+	 $(this).prop("disabled",true);
+	 tournamentBusiness.startTournament(validationResult);
```

Comment:
So are there errors where you start tournament? This is hard to follow, and also that validation result is processed partially here and partially inside. Try to keep logic tight.